### DISCUSSION TITLE
Anchoring a 3d object in Hololens

### DISCUSSION THREAD

**Discussion by tharakanandakumara** [StereoKit/StereoKit#161] [Q&A] [answered]
Is there a way we can overlay a 3d object on a specific location in HoloLens? For example overlaying a 3d object of a window on actual window through HoloLens.

**Comment by maluoi** [comment, marked as answer]
This kinda depends on how you're trying to do it! Displaying it is easy, locating it could be very tricky, persisting it across sessions should not be too hard.

To locate it, you could use things like [QR codes](https://stereokit.net/Pages/Guides/Using-QR-Codes.html) for placement, [Scene Understanding](https://docs.microsoft.com/en-us/windows/mixed-reality/develop/platform-capabilities-and-apis/scene-understanding-sdk) might tell you about the wall it's on, you could let the user place it by hand, or perhaps do some machine learning stuff via the camera? (Scene Understanding will be coming natively to StereoKit in v0.3.2)

To persist across session, there's [world anchors](https://gist.github.com/maluoi/0671623b02dbe5a11997eb769e5ea635), [Azure Spatial Anchors](https://docs.microsoft.com/en-us/azure/spatial-anchors/), and QR codes again.

And display is just rendering something at that location, which is as easy as a Draw call with the mesh/model you've got :)

**Reply by tharakanandakumara** [reply]
Thanks. If I choose QR code approach, Is there a way which I can get location of multiple QR codes in HoloLens 3D space and place my 3D object using those multiple coordinates?

**Reply by maluoi** [reply]
Yeah, absolutely! With the QR code sample I linked to, it'll give you a list of all QR codes it finds. Each one has a (pretty accurate!) position and orientation, and the data contained on the QR code itself. That should be plenty to identify, average, or whatever else you want to do with 'em.
